pkg/tsdb/cloudwatch/response_parser_test.go

```diff
@@ -133,6 +139,83 @@ func TestCloudWatchResponseParser(t *testing.T) {
 			So(timeSeries2.Tags["LoadBalancer"], ShouldEqual, "lb2")
 		})
 
+		Convey("can expand alias when there's no data returned", func() {
+			resp := map[string]*cloudwatch.MetricDataResult{}
+			query := &cloudWatchQuery{
+				RefId:      "refId1",
+				Region:     "us-east-1",
+				Namespace:  "AWS/ApplicationELB",
+				MetricName: "TargetResponseTime",
+				Dimensions: map[string][]string{
+					"AvailabilityZone": {"az1", "az2"},
+					"LoadBalancer":     {"lb1", "lb2"},
+					"TargetGroup":      {"tg"},
+				},
+				Stats:  "Average",
+				Period: 60,
+				Alias:  "{{AvailabilityZone}} Expanded",
+			}
+			series, partialData, err := parseGetMetricDataTimeSeries(resp, query)
+
+			timeSeries := (*series)[0]
+			So(err, ShouldBeNil)
+			So(partialData, ShouldBeFalse)
+			So(timeSeries.Name, ShouldEqual, "az1 Expanded")
+			So(timeSeries.Tags["AvailabilityZone"], ShouldEqual, "az1")
+
+			timeSeries2 := (*series)[1]
+			So(timeSeries2.Name, ShouldEqual, "az2 Expanded")
+			So(timeSeries2.Tags["AvailabilityZone"], ShouldEqual, "az2")
+
+		})
+
+		Convey("can expand alias when there's partial az data returned", func() {
+			timestamp := time.Unix(0, 0)
+			resp := map[string]*cloudwatch.MetricDataResult{
+				"az1": {
+					Id:    aws.String("id1"),
+					Label: aws.String("az1"),
+					Timestamps: []*time.Time{
+						aws.Time(timestamp),
+						aws.Time(timestamp.Add(60 * time.Second)),
+						aws.Time(timestamp.Add(180 * time.Second)),
+					},
+					Values: []*float64{
+						aws.Float64(10),
+						aws.Float64(20),
+						aws.Float64(30),
+					},
+					StatusCode: aws.String("Complete"),
+				},
+			}
+			query := &cloudWatchQuery{
+				RefId:      "refId1",
+				Region:     "us-east-1",
+				Namespace:  "AWS/ApplicationELB",
+				MetricName: "TargetResponseTime",
+				Dimensions: map[string][]string{
+					"AvailabilityZone": {"az1", "az2"},
```

Comment:
The test will fail if you remove one of the availability zones.

Reply:
Can take a look at that!